src/generate.dart

```diff
@@ -74,90 +85,133 @@ class DiagramGenerator {
 
   /// The temporary directory used to write screenshots and cropped out images
   /// into.
-  Directory tmpDir;
+  Directory temporaryDirectory;
 
-  static String get projectDir =>
-      path.joinAll(path.split(path.fromUri(Platform.script))..removeLast());
+  static String get projectDir {
+    return path.joinAll(path.split(path.fromUri(Platform.script))..removeLast());
+  }
 
   Future<Null> generateDiagram() async {
     await _collectScreenshot();
     await _optimizeImages();
     if (cleanup) {
-      await tmpDir.delete(recursive: true);
+      await temporaryDirectory.delete(recursive: true);
     }
   }
 
-  bool _captureScreenshot(String outputName) {
-    print('Capturing screenshot into $outputName.');
-    ProcessResult processResult = processRunner(flutterCommand, ['screenshot', '--out=$outputName'],
-        workingDirectory: projectDir);
-    if (processResult.exitCode != 0) {
-      print("Failed to run command '$flutterCommand screenshot --out=$outputName' in ${tmpDir.path}:\n${processResult.stderr}");
-      return false;
+  static const Duration timeout = const Duration(seconds: 60);
+
+  Future<List<String>> runFlutter({
+    @required List<String> args,
+    @required String until,
+    AsyncCallback then,
+  }) async {
+    final Process process = await processStarter(flutterCommand, args, workingDirectory: projectDir);
+    final List<String> lines = <String>[];
+    process.stdout.transform(UTF8.decoder).transform(const LineSplitter()).listen((String data) {
+      lines.add(data);
+    });
+    process.stderr.transform(UTF8.decoder).transform(const LineSplitter()).listen((String data) {
+      lines.add(data);
+    });
+    final DateTime startWait = new DateTime.now();
+    while (!lines.any((String line) => line.contains(until))) {
+      if (new DateTime.now().difference(startWait) > timeout) {
+        stdout.write('\n');
+        print('Timed out.');
+        print('Output from application:');
+        lines.forEach(print);
```

Comment:
Wouldn't you want to write stdout to stdout (via print), and stderr to stderr?

Reply:
What would the functional difference be? I mean it all just ends up on the console, right.

Reply:
If someone redirects output, it'll go different places.  It's not hugely important in this case.